### PR TITLE
[iotedge check] bugbash fixes

### PR DESCRIPTION
**Pull request by daprilik**
- Temporarily remove aziot-edged version check until we sort out where the version info for 1.2 will live
- Remove references to config.yaml, replace with broader "configuration" terminology

### CODE REVIEW

File: edgelet/iotedge/src/check/checks/well_formed_config.rs [hunk 1]
```diff
@@ -12,10 +12,10 @@ pub(crate) struct WellFormedConfig {}
 
 impl Checker for WellFormedConfig {
     fn id(&self) -> &'static str {
-        "config-yaml-well-formed"
+        "config-toml-well-formed"
```
Comment (arsing) [inline edgelet/iotedge/src/check/checks/well_formed_config.rs:15]: It's not a single TOML here either. Just make it `config` and `"configuration"` like in your i-i-s PR.

Comment (daprilik) [reply edgelet/iotedge/src/check/checks/well_formed_config.rs]: Yep, sorry, that's my pre-lunch hunger manifesting itself.

**Review by arsing — APPROVED** [review]

**Comment by arsing** [comment]
A clippy failure because the field read by the latest-version check is no longer used.

**Comment by daprilik** [comment]
I decided to disable the check "from the inside", which keeps all the changes / clippy annotations limited to one file, thereby making it easy to revert the change in the future.

**Review by arsing — APPROVED** [review]

**Pull request merged**